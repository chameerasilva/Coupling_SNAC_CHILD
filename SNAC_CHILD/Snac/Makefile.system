ifndef WHICH
	WHICH=/usr/bin/which
endif
ifndef UNAME
	UNAME=/usr/bin/uname
endif
ifndef UNAME_M
	UNAME_M=/usr/bin/uname -m
endif
ifndef UNAME_R
	UNAME_R=/usr/bin/uname -r
endif
ifndef MACHINE
	MACHINE=x86_64
endif
ifndef KERNEL_RELEASE
	KERNEL_RELEASE=OtherCats
endif
ifndef UNAME_S
	UNAME_S=/usr/bin/uname -s
endif
ifndef SYSTEM
	SYSTEM=Darwin
endif
ifndef MAKE
	MAKE=/usr/bin/make
endif
ifndef SH
	SH=/bin/sh
endif
ifeq (default,$(origin RM))
	RM=/bin/rm
endif
ifndef RM
	RM=/bin/rm
endif
ifndef RM_F
	RM_F=/bin/rm -rf
endif
ifndef CD
	CD=cd
endif
ifndef CP
	CP=/bin/cp
endif
ifndef CP_F
	CP_F=/bin/cp -f
endif
ifndef MV
	MV=/bin/mv
endif
ifndef MV_F
	MV_F=/bin/mv -f
endif
ifndef MKDIR
	MKDIR=mkdir -p
endif
ifndef SED
	SED=/usr/bin/sed
endif
ifndef FIND
	FIND=/usr/bin/find
endif
ifndef LN
	LN=/bin/ln
endif
ifndef LN_S
	LN_S=/bin/ln -fs
endif
ifndef TEE
	TEE=/usr/bin/tee
endif
ifndef NM
	NM=/opt/local/bin/nm
endif
ifndef LS
	LS=/bin/ls
endif
ifndef GREP
	GREP=/usr/bin/grep
endif
ifndef AR
	AR=/usr/bin/ar
endif
ifndef AR_C
	AR_C=${AR} -cru
endif
ifndef RANLIB
	RANLIB=/usr/bin/ranlib
endif
ifeq (default,$(origin CC))
	CC=/usr/bin/cc
endif
ifndef CC
	CC=/usr/bin/cc
endif
ifndef CC_TYPE
	CC_TYPE=gnu
endif
ifeq (default,$(origin CXX))
	CXX=/usr/bin/g++
endif
ifndef CC_PIPE
	CC_PIPE=-pipe
endif
ifndef CC_64
	CC_64=-m64
endif
ifndef SYSTEM_SIZEOF_LONG
	SYSTEM_SIZEOF_LONG=8
endif
ifndef SYSTEM_64
	SYSTEM_64=1
endif
ifndef SYSTEM_BIG_ENDIAN
	SYSTEM_BIG_ENDIAN=0
endif
ifndef OBJCC
	OBJCC=/usr/bin/cc -x objective-c
endif
ifndef OBJECTIVEC_LIBS
	OBJECTIVEC_LIBS=-lobjc
endif
ifndef CXX_EXT
	CXX_EXT=cxx
endif
ifndef CXX
	CXX=/usr/bin/g++
endif
ifndef CXX_TYPE
	CXX_TYPE=gnu
endif
ifndef CXX_PIPE
	CXX_PIPE=-pipe
endif
ifeq (default,$(origin F77))
	F77=
endif
ifndef F77
	F77=
endif
ifndef F77_TYPE
	F77_TYPE=
endif
ifndef LINKER
	LINKER=/usr/bin/cc
endif
ifndef SO_EXT
	SO_EXT=dylib
endif
ifndef MODULE_EXT
	MODULE_EXT=so
endif
ifndef SO_CFLAGS
	SO_CFLAGS=
endif
ifndef SO_LFLAGS
	SO_LFLAGS=-dynamiclib -flat_namespace -undefined suppress -single_module
endif
ifndef MODULE_LFLAGS
	MODULE_LFLAGS=-bundle -flat_namespace -undefined suppress
endif
ifndef RPATH_FLAG
	RPATH_FLAG=
endif
ifndef RPATH
	RPATH=
endif
ifndef RPATH_LFLAGS
	RPATH_LFLAGS=
endif
ifndef EXPORT_DYNAMIC_LFLAGS
	EXPORT_DYNAMIC_LFLAGS=-framework System -framework CoreServices -framework Foundation
endif
ifndef CC_CXX_LFLAGS
	CC_CXX_LFLAGS=-lstdc++
endif
ifndef FORTRAN_COMPILER_LIBDIR
	FORTRAN_COMPILER_LIBDIR=
endif
ifndef EXTRA_FORTRAN_LIBS
	EXTRA_FORTRAN_LIBS=
endif
ifndef CC_FORTRAN_LFLAGS
	CC_FORTRAN_LFLAGS=
endif
ifndef F77_LIBS
	F77_LIBS=
endif
ifndef F77_SYMBOL_TRANSLATION
	F77_SYMBOL_TRANSLATION=
endif
ifndef CC_WARNINGLEVEL
	CC_WARNINGLEVEL=-Wall
endif
ifndef CC_SYMBOLLEVEL
	CC_SYMBOLLEVEL= 
endif
ifndef CC_OPTIMISATIONLEVEL
	CC_OPTIMISATIONLEVEL=-O3 -march=athlon64 -DNDEBUG
endif
ifndef NOSHARED
	NOSHARED=
endif
ifndef DL_DIR
	DL_DIR=/sw
endif
ifndef DL_LIBDIR
	DL_LIBDIR=/sw/lib
endif
ifndef DL_LIBFILES
	DL_LIBFILES=
endif
ifndef DL_LIBS
	DL_LIBS=-L${DL_LIBDIR} ${DL_LIBFILES}
endif
ifndef DL_INCDIR
	DL_INCDIR=/sw/include
endif
ifndef DL_INCLUDES
	DL_INCLUDES=-I/sw/include
endif
ifndef NOPYTHON
	NOPYTHON=
endif
ifndef DOXYGEN
	DOXYGEN=/usr/bin/doxygen
endif
ifndef SVN
	SVN=/opt/local/bin/svn
endif
ifndef HAVE_SVN
	HAVE_SVN=1
endif
ifndef CSOAP_DIR
	CSOAP_DIR=/usr/lib/libcsoap
endif
ifndef CSOAP_INCDIR
	CSOAP_INCDIR=${CSOAP_DIR}/include/libcsoap-1.0
endif
ifndef CSOAP_INCLUDES
	CSOAP_INCLUDES=-I${CSOAP_INCDIR}
endif
ifndef CSOAP_LIBDIR
	CSOAP_LIBDIR=${CSOAP_DIR}/lib
endif
ifndef CSOAP_LIBFILES
	CSOAP_LIBFILES=-lcsoap-1.0
endif
ifndef CSOAP_LIBS
	CSOAP_LIBS=-L${CSOAP_LIBDIR} ${CSOAP_LIBFILES}
endif
ifndef NANOHTTP_DIR
	NANOHTTP_DIR=${CSOAP_DIR}
endif
ifndef NANOHTTP_INCDIR
	NANOHTTP_INCDIR=${NANOHTTP_DIR}/include/nanohttp-1.0
endif
ifndef NANOHTTP_INCLUDES
	NANOHTTP_INCLUDES=-I${NANOHTTP_INCDIR}
endif
ifndef NANOHTTP_LIBDIR
	NANOHTTP_LIBDIR=${NANOHTTP_DIR}/lib
endif
ifndef NANOHTTP_LIBFILES
	NANOHTTP_LIBFILES=-lnanohttp-1.0
endif
ifndef NANOHTTP_LIBS
	NANOHTTP_LIBS=-L${NANOHTTP_LIBDIR} ${NANOHTTP_LIBFILES}
endif
ifndef HAVE_SOAP
	HAVE_SOAP=
endif
ifndef REGRESSTOR_URL
	REGRESSTOR_URL=http://localhost/~alan/cgi-bin/RegresstorServices/Regresstor.cgi
endif
USE_OPTIMISED=1
ifndef STGERMAIN_DIR
	STGERMAIN_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised
endif
ifndef OPTIONS
	OPTIONS=optimised
endif
ifndef USE_RUN_LONG_TESTS
	USE_RUN_LONG_TESTS=
endif
ifndef PROJ_ROOT
	PROJ_ROOT=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/Snac
endif
ifndef BLD_DIR
	BLD_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised
endif
ifndef EXPORT_DIR
	EXPORT_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised
endif
ifndef TMP_DIR
	TMP_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/tmp
endif
ifndef BIN_DIR
	BIN_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/bin
endif
ifndef LIB_DIR
	LIB_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/lib
endif
ifndef INC_DIR
	INC_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/include
endif
ifndef TST_DIR
	TST_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/tests
endif
ifndef PYB_DIR
	PYB_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/Python
endif
ifndef PYC_DIR
	PYC_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/Python
endif
ifndef SHARE_DIR
	SHARE_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/share
endif
PROJECT=Snac
ifndef FIND_SUBDIRS
	FIND_SUBDIRS=/usr/bin/find * -type d -follow -prune ! -name .svn
endif
ifndef MAKEFILE_FIND_SUBDIRS
	MAKEFILE_FIND_SUBDIRS=$(shell ${FIND_SUBDIRS})
endif
ifndef GET_MODNAME
	GET_MODNAME=basename /Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/StGermain
endif
ifndef MAKEFILE_GET_MODNAME
	MAKEFILE_GET_MODNAME=$(shell ${GET_MODNAME})
endif
ifndef MAKEFILE_CHECK_DEPENDENCIES
	MAKEFILE_CHECK_DEPENDENCIES=$(foreach DEP,${DEPENDS},$(shell cd ${LIB_DIR} && if test -z `find . -name ${PROJECT}${DEP}*` ; then echo Compiling dependency: ${DEP} 1>&2 && cd ${PROJ_ROOT}/src && if test ! -z `find . -type d -name ${DEP}` ; then cd `find . -type d -name ${DEP}` && make 1>&2 ; else echo Cannot find dependency: ${DEP} 1>&2 ; fi ; fi ) )
endif
ifndef CC_SYMBOLLEVEL
	CC_SYMBOLLEVEL= 
endif
ifndef CC_WARNINGLEVEL
	CC_WARNINGLEVEL=-Wall
endif
ifndef CC_DEBUGLEVEL
	CC_DEBUGLEVEL=-DDEBUG -DQUERY_FP
endif
ifndef MATH_DIR
	MATH_DIR=/usr
endif
ifndef MATH_INCDIR
	MATH_INCDIR=/usr/include
endif
ifndef MATH_INCLUDES
	MATH_INCLUDES=-I/usr/include
endif
ifndef MATH_LIBDIR
	MATH_LIBDIR=/usr/lib
endif
ifndef MATH_LIBS
	MATH_LIBS=-L/usr/lib -lm -lmx
endif
ifndef MPI_DIR
	MPI_DIR=/Users/eunseo/opt/openmpi-1.6.1
endif
ifndef MPI_BINDIR
	MPI_BINDIR=/Users/eunseo/opt/openmpi-1.6.1/bin
endif
ifndef MPI_LIBDIR
	MPI_LIBDIR=/Users/eunseo/opt/openmpi-1.6.1/lib
endif
ifndef MPI_INCDIR
	MPI_INCDIR=/Users/eunseo/opt/openmpi-1.6.1/include
endif
ifndef MPI_INCLUDES
	MPI_INCLUDES=-I/Users/eunseo/opt/openmpi-1.6.1/include
endif
ifndef MPI_IMPLEMENTATION
	MPI_IMPLEMENTATION=open_mpi
endif
ifndef MPI_LIBFILES
	MPI_LIBFILES=-lmpi
endif
ifndef MPI_RPATH
	MPI_RPATH=
endif
ifndef MPI_LIBS
	MPI_LIBS=-L/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/lib -lmpi -L/Users/eunseo/opt/openmpi-1.6.1/lib -lmpi -L/sw/lib 
endif
ifndef MPI_RUN_COMMAND
	MPI_RUN_COMMAND=mpirun
endif
ifndef MPI_RUN
	MPI_RUN=/Users/eunseo/opt/openmpi-1.6.1/bin/mpirun
endif
ifndef MPI_NPROC
	MPI_NPROC=-np
endif
HAVE_MPI=1
ifndef XML2_CONFIG
	XML2_CONFIG=/Users/eunseo/anaconda/bin/xml2-config
endif
ifndef XML_DIR
	XML_DIR=/Users/eunseo/anaconda
endif
ifndef XML_CFLAGS
	XML_CFLAGS=-I/Users/eunseo/anaconda/include/libxml2
endif
ifndef XML_INCLUDES
	XML_INCLUDES=-I/Users/eunseo/anaconda/include/libxml2
endif
ifndef XML_LIBS
	XML_LIBS=-L/Users/eunseo/anaconda/lib -lxml2 -lz -liconv -lm
endif
ifndef STGERMAIN_DIR
	STGERMAIN_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised
endif
ifndef STGERMAIN_INCDIR
	STGERMAIN_INCDIR=${STGERMAIN_DIR}/include
endif
ifndef STGERMAIN_INCLUDES
	STGERMAIN_INCLUDES=-I${STGERMAIN_INCDIR}
endif
ifndef STGERMAIN_LIBDIR
	STGERMAIN_LIBDIR=${STGERMAIN_DIR}/lib
endif
ifndef STGERMAIN_LIBS
	STGERMAIN_LIBS=-L${STGERMAIN_LIBDIR} -lStGermain
endif
ifndef STGERMAIN_RPATH
	STGERMAIN_RPATH=
endif
ifndef HAVE_STGERMAIN
	HAVE_STGERMAIN=1
endif
ifndef APPS_XML_DIR
	APPS_XML_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/lib/SNAC
endif
	EXPORT_INCLUDE_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/lib/StGermain/Snac
	EXAMPLES_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/share/Snac
	STG_INCLUDE_PATH=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/lib/StGermain
	EXPORT_INCLUDE_DIR=/Users/eunseo/projects/SNAC_CHILD_Coupling/SNAC_CHILD/build-optimised/lib/StGermain/Snac
