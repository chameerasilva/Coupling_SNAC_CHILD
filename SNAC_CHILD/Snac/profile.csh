##~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
##
## Copyright (C), 2003, Victorian Partnership for Advanced Computing (VPAC) Ltd, 110 Victoria Street, Melbourne, 3053, Australia.
##
## Authors:
##	Stevan M. Quenette, Senior Software Engineer, VPAC. (steve@vpac.org)
##
## This program is free software; you can redistribute it and/or modify it
## under the terms of the GNU General Public License as published by the
## Free Software Foundation; either version 2, or (at your option) any
## later version.
## 
## This program is distributed in the hope that it will be useful,
## but WITHOUT ANY WARRANTY; without even the implied warranty of
## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
## GNU General Public License for more details.
## 
## You should have received a copy of the GNU General Public License
## along with this program; if not, write to the Free Software
## Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
##
## Role:
##	Set up a csh environment for using StGermain built things.
##
## $Id: profile.csh,v 1.2 2003/11/17 09:17:31 steve Exp $
##
##~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

setenv PATH `pwd`/build/bin:$PATH
setenv PYTHONPATH `pwd`/build/Python:$PYTHONPATH
setenv LD_LIBRARY_PATH `pwd`/build/lib:$LD_LIBRARY_PATH
setenv DYLD_LIBRARY_PATH `pwd`/build/lib:$DYLD_LIBRARY_PATH
