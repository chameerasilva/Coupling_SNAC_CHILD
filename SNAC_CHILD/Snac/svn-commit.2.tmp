GSL not needed any more - don't do the GSL configuration.
Barycentric Interpolation (BI) is now the remeshing scheme of choice.
--This line, and those below, will be ignored--

M    project-config.sh
M    plugins/Makefile.def
