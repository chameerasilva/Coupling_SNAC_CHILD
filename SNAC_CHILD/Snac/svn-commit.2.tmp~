
--This line, and those below, will be ignored--

M    project-config.sh
M    plugins/Makefile.def
