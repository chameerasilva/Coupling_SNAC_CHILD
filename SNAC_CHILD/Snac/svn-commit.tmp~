
--This line, and those below, will be ignored--

M    libSnac/src/TetrahedraTables.c
M    libSnac/src/TetrahedraTables.h
M    plugins/remesher_BI/RemeshElements.c
M    plugins/remesher_BI/Mesh.h
M    plugins/remesher_BI/Utils.h
M    plugins/remesher_BI/Remesh.c
M    plugins/remesher_BI/makefile
M    plugins/plastic_BI/Remesh.c
