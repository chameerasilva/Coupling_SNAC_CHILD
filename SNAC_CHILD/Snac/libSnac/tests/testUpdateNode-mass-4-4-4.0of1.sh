#!/bin/sh

TEST_SCRIPT=./VMake/executableTester.sh
until test -r ${TEST_SCRIPT} ; do
        TEST_SCRIPT=../${TEST_SCRIPT}
done
. ${TEST_SCRIPT}

runAndHandleSystemTest "testUpdateNode-mass-4-4-4 data/default.xml" "$0" "$@"
