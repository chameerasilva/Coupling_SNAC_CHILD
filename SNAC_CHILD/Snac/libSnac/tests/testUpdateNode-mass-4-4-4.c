/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
**
** Copyright (C), 2003, 
**	Steve Quenette, 110 Victoria Street, Melbourne, Victoria, 3053, Australia.
**	Californian Institute of Technology, 1200 East California Boulevard, Pasadena, California, 91125, USA.
**	University of Texas, 1 University Station, Austin, Texas, 78712, USA.
**
** Authors:
**	Stevan M. Quenette, Senior Software Engineer, VPAC. (steve@vpac.org)
**	Stevan M. Quenette, Visitor in Geophysics, Caltech.
**	Luc Lavier, Research Scientist, The University of Texas. (luc@utig.ug.utexas.edu)
**	Luc Lavier, Research Scientist, Caltech.
**
** This program is free software; you can redistribute it and/or modify it
** under the terms of the GNU General Public License as published by the
** Free Software Foundation; either version 2, or (at your option) any
** later version.
** 
** This program is distributed in the hope that it will be useful,
** but WITHOUT ANY WARRANTY; without even the implied warranty of
** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
** GNU General Public License for more details.
** 
** You should have received a copy of the GNU General Public License
** along with this program; if not, write to the Free Software
** Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
**
** Role:
**	Tests StrainRate function for values of element volume.
**
** $Id: testUpdateNode-mass-4-4-4.c 1792 2004-07-30 05:42:39Z SteveQuenette $
**
**~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/

#include "mpi.h"
#include <StGermain/StGermain.h>
#include <StGermain/FD/FD.h>
#include "Snac/Snac.h"
#include "TestUtilities.h"

#include "stdio.h"
#include "stdlib.h"


extern MPI_Comm			CommWorld;
extern int				rank;
extern int				numProcessors;
extern int				procToWatch;
extern Dictionary*			dictionary;
extern Snac_Context*			snacContext;

int main( int argc, char* argv[] ) {
	Node_LocalIndex		nodeIndex;
	double				speedOfSound;
	Mass				mass000, mass100, mass110, mass111;
	Mass				inertialMass000, inertialMass100, inertialMass110, inertialMass111;
	Force				force000, force100, force110, force111;
	Force				balance000, balance100, balance110, balance111;
	
	SnacTest_SetUp( argc, argv );

	SnacTest_PrintElementZeroInfo();
	
	SnacTest_UpdateElementsNodes( &speedOfSound );
	
	/* obtain the first corner node's info */
	nodeIndex = RegularMeshUtils_Node_Local3DTo1D( (HexaMD*)snacContext->mesh->layout->decomp, 0, 0, 0 );
	Snac_Force( (Context*)snacContext, nodeIndex, speedOfSound, &mass000, &inertialMass000, &force000, &balance000 );
	Snac_UpdateNodeMomentum( (Context*)snacContext, nodeIndex, inertialMass000, force000 );
	printf( "Corner   Node %3u (0,0,0): mass: %g\n", nodeIndex, mass000 );
	
	/* obtain the first edge node's info */
	nodeIndex = RegularMeshUtils_Node_Local3DTo1D( (HexaMD*)snacContext->mesh->layout->decomp, 1, 0, 0 );
	Snac_Force( (Context*)snacContext, nodeIndex, speedOfSound, &mass100, &inertialMass100, &force100, &balance100 );
	Snac_UpdateNodeMomentum( (Context*)snacContext, nodeIndex, inertialMass100, force100 );
	printf( "Edge     Node %3u (1,0,0): mass: %g\n", nodeIndex, mass100 );
	
	/* obtain the first suface node's info */
	nodeIndex = RegularMeshUtils_Node_Local3DTo1D( (HexaMD*)snacContext->mesh->layout->decomp, 1, 1, 0 );
	Snac_Force( (Context*)snacContext, nodeIndex, speedOfSound, &mass110, &inertialMass110, &force110, &balance110 );
	Snac_UpdateNodeMomentum( (Context*)snacContext, nodeIndex, inertialMass110, force110 );
	printf( "Surface  Node %3u (1,1,0): mass: %g\n", nodeIndex, mass110 );
	
	/* obtain the first internal node's info */
	nodeIndex = RegularMeshUtils_Node_Local3DTo1D( (HexaMD*)snacContext->mesh->layout->decomp, 1, 1, 1 );
	Snac_Force( (Context*)snacContext, nodeIndex, speedOfSound, &mass111, &inertialMass111, &force111, &balance111 );
	Snac_UpdateNodeMomentum( (Context*)snacContext, nodeIndex, inertialMass111, force111 );
	printf( "Internal Node %3u (1,1,1): mass: %g\n", nodeIndex, mass111 );
	
	
	/* For each node, compare to the appropriate node */
	for( nodeIndex = 1; nodeIndex < snacContext->mesh->nodeLocalCount; nodeIndex++ ) {
		IJK		ijk;
		const IJK	s = { 	((HexaMD*)snacContext->mesh->layout->decomp)->nodeLocal3DCounts[rank][0] - 1,
					((HexaMD*)snacContext->mesh->layout->decomp)->nodeLocal3DCounts[rank][1] - 1,
					((HexaMD*)snacContext->mesh->layout->decomp)->nodeLocal3DCounts[rank][2] - 1 };
		
		RegularMeshUtils_Node_Local1DTo3D( (HexaMD*)snacContext->mesh->layout->decomp, nodeIndex, &ijk[0], &ijk[1], 
			&ijk[2] );
		if( (ijk[0] == 0 || ijk[0] == s[0]) && (ijk[1] == 0 || ijk[1] == s[1]) && (ijk[2] == 0 || ijk[2] == s[2]) ) {
			Bool		error;
			Mass		mass;
			Mass		inertialMass;
			Force		force;
			Force		balance;
			
			error = False;		
			Snac_Force( (Context*)snacContext, nodeIndex, speedOfSound, &mass, &inertialMass, &force, &balance );
			Snac_UpdateNodeMomentum( (Context*)snacContext, nodeIndex, inertialMass, force );
			if( mass != mass000 ) {
				printf( "Corner   Node %3u: mass: %g\n", nodeIndex, mass );
				error = True;
			}
			
			if( error ) {
				printf( "Corner   Node %3u: Gives a differing answer from node {0, 0, 0}.\n", nodeIndex );
			}
			else {
				printf( "Corner   Node %3u: has same values as node {0, 0, 0}.\n", nodeIndex );
			}
		}
		else if( ((ijk[0] == 0 || ijk[0] == s[0]) && (ijk[1] == 0 || ijk[1] == s[1])) ||
			((ijk[0] == 0 || ijk[0] == s[0]) && (ijk[2] == 0 || ijk[2] == s[2])) ||
			((ijk[1] == 0 || ijk[1] == s[1]) && (ijk[2] == 0 || ijk[2] == s[2])) )
		{
			Bool 		error;
			Mass		mass;
			Mass		inertialMass;
			Force		force;
			Force		balance;
			
			error = False;		
			Snac_Force( (Context*)snacContext, nodeIndex, speedOfSound, &mass, &inertialMass, &force, &balance );
			Snac_UpdateNodeMomentum( (Context*)snacContext, nodeIndex, inertialMass, force );
			if( mass != mass100 ) {
				printf( "Edge     Node %3u: mass: %g\n", nodeIndex, mass );
				error = True;
			}
			
			if( error ) {
				printf( "Edge     Node %3u: Gives a differing answer from node {1, 0, 0}.\n", nodeIndex );
			}
			else {
				printf( "Edge     Node %3u: has same values as node {1, 0, 0}.\n", nodeIndex );
			}
		}
		else if( (ijk[0] == 0 || ijk[0] == s[0]) || (ijk[1] == 0 || ijk[1] == s[1]) || (ijk[2] == 0 || ijk[2] == s[2]) ) {
			Bool 		error;
			Mass		mass;
			Mass		inertialMass;
			Force		force;
			Force		balance;
			
			error = False;		
			Snac_Force( (Context*)snacContext, nodeIndex, speedOfSound, &mass, &inertialMass, &force, &balance );
			Snac_UpdateNodeMomentum( (Context*)snacContext, nodeIndex, inertialMass, force );
			if( mass != mass110 ) {
				printf( "Surface  Node %3u: mass: %g\n", nodeIndex, mass );
				error = True;
			}
			
			if( error ) {
				printf( "Surface  Node %3u: Gives a differing answer from node {1, 1, 0}.\n", nodeIndex );
			}
			else {
				printf( "Surface  Node %3u: has same values as node {1, 1, 0}.\n", nodeIndex );
			}
		}
		else {
			Bool 		error;
			Mass		mass;
			Mass		inertialMass;
			Force		force;
			Force		balance;
			
			error = False;		
			Snac_Force( (Context*)snacContext, nodeIndex, speedOfSound, &mass, &inertialMass, &force, &balance );
			Snac_UpdateNodeMomentum( (Context*)snacContext, nodeIndex, inertialMass, force );
			if( mass != mass111 ) {
				printf( "Internal Node %3u: mass: %g\n", nodeIndex, mass );
				error = True;
			}
			
			if( error ) {
				printf( "Internal Node %3u: Gives a differing answer from node {1, 1, 1}.\n", nodeIndex );
			}
			else {
				printf( "Internal Node %3u: has same values as node {1, 1, 1}.\n", nodeIndex );
			}
		}
	}
	
	SnacTest_TearDown();
	
	return 0; /* success */
}
