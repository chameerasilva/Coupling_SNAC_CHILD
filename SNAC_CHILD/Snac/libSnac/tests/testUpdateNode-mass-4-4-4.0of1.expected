StGermain Framework revision 3890. Copyright (C) 2003-2005 VPAC.
StGermain Discretisation Library revision 3890. Copyright (C) 2003-2005 VPAC.
Snac Framework. Copyright (C) 2003-2005 Caltech, VPAC & University of Texas.
Watching rank: 0
"dtType" set by Dictionary to "constant"
"forceCalcType" set by Dictionary to "complete"
Constructing context..
	
	Creating Stg_Components from the component-list
	
	
	Constructing Stg_Components from the live-component register
	
For Material 0:
	rheology = 1
	alpha = 0.000000e+00
	beta = 0.000000e+00

	lambda = 3.000000e+10
	mu = 3.000000e+10

	maxiterations = 1
	constitutivetolerance = 1.000000e-03
	yieldcriterion = 0
	nsegments = 2
		seg 0: plstrain = 0.000000e+00
		seg 0: frictionAngle = 0.000000e+00
		seg 0: dilationAngle = 0.000000e+00
		seg 0: cohesion = 0.000000e+00
		seg 1: plstrain = 0.000000e+00
		seg 1: frictionAngle = 0.000000e+00
		seg 1: dilationAngle = 0.000000e+00
		seg 1: cohesion = 0.000000e+00
		seg 2: plstrain = 0.000000e+00
		seg 2: frictionAngle = 0.000000e+00
		seg 2: dilationAngle = 0.000000e+00
		seg 2: cohesion = 0.000000e+00
	ten_off = 0.000000e+00
	puSeeds = 0

	vis_min = 1.000000e+18
	vis_max = 3.000000e+27
	refvisc = 1.000000e+19
	refsrate = 1.000000e-15
	reftemp = 1.400000e+03
	activationE = 5.400000e+03
	srexponent = 1.000000e+00
	srexponent1 = 1.000000e+00
	srexponent2 = 1.000000e+00

	thermal conductivity = 2.000000e+00
	heat capacity = 1.000000e+03
	density = 2.700000e+03
In: Snac_Context_TimeStepZero
self->timeStep: 0 (update elements only)
self->currentTime: 0
Element: 0, Coords: (0 -1 0), (0.5 -1 0), (0 -0.5 0), (0.5 -0.5 0), (0 -1 0.5) (0.5 -1 0.5) (0 -0.5 0.5) (0.5 -0.5 0.5)
Element: 0, Vel: (0 0 0), (0 0 0), (0 0 0), (0 0 0), (0 0 0) (0 0 0) (0 0 0) (0 0 0)
Global minLengthScale: 0.096225
Dt: 1
Speed of sound: 0.0481125
Corner   Node   0 (0,0,0): mass: 42.1875
Edge     Node   1 (1,0,0): mass: 84.375
Surface  Node   4 (1,1,0): mass: 168.75
Internal Node  13 (1,1,1): mass: 337.5
Edge     Node   1: has same values as node {1, 0, 0}.
Corner   Node   2: has same values as node {0, 0, 0}.
Edge     Node   3: has same values as node {1, 0, 0}.
Surface  Node   4: has same values as node {1, 1, 0}.
Edge     Node   5: has same values as node {1, 0, 0}.
Corner   Node   6: has same values as node {0, 0, 0}.
Edge     Node   7: has same values as node {1, 0, 0}.
Corner   Node   8: has same values as node {0, 0, 0}.
Edge     Node   9: has same values as node {1, 0, 0}.
Surface  Node  10: has same values as node {1, 1, 0}.
Edge     Node  11: has same values as node {1, 0, 0}.
Surface  Node  12: has same values as node {1, 1, 0}.
Internal Node  13: has same values as node {1, 1, 1}.
Surface  Node  14: has same values as node {1, 1, 0}.
Edge     Node  15: has same values as node {1, 0, 0}.
Surface  Node  16: has same values as node {1, 1, 0}.
Edge     Node  17: has same values as node {1, 0, 0}.
Corner   Node  18: has same values as node {0, 0, 0}.
Edge     Node  19: has same values as node {1, 0, 0}.
Corner   Node  20: has same values as node {0, 0, 0}.
Edge     Node  21: has same values as node {1, 0, 0}.
Surface  Node  22: has same values as node {1, 1, 0}.
Edge     Node  23: has same values as node {1, 0, 0}.
Corner   Node  24: has same values as node {0, 0, 0}.
Edge     Node  25: has same values as node {1, 0, 0}.
Corner   Node  26: has same values as node {0, 0, 0}.
